crates/matrix-sdk/src/widget/mod.rs

```diff
@@ -1,13 +1,15 @@
-//! Client widget API implementation.
+//! Widget API implementation.
 
-use async_channel::{Receiver, Sender};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
```

Comment:
Why was this reverted to tokio's channel types?

Reply:
I'd say it way by accident (a byproduct of rushed creation of the PR - I replaced the `mod.rs` at some point and did not rigorously check the diff). I think we could use `async_channel` for sure, however I have a question related to it: what's the advantage of `async_channel` in our case? - Do we really want to have *multiple consumers*? My primary concerns are:

1. It basically states on the API level that it's possible to have multiple consumers. This potentially means that someone who constructs a `Comm` channel may have created multiple consumers, which means that certain messages may be "intercepted" by different consumers (only one of the existing consumers gets the message), whereas with the `mpsc` we know that it's only one consumer (our client widget API implementation) that gets and handles the message. It feels like having multiple consumers is rather useful if we designed some sort of a "worker pool" where one worker gets a task and executes it.
2. Unlike Tokio's `mpsc`, the `send()` method returns a future (even if the channel is unbounded), i.e. we're expected to `await` on it (even though it should not be necessary for the unbounded channels).